hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -205,6 +215,26 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
   }
 
 
+  /**
+   * Add default acls for bucket. These acls are inherited from volume
+   * default acl list.
+   * @param omBucketInfo
+   * @param omVolumeArgs
+   */
+  private void addDefaultAcls(OmBucketInfo omBucketInfo,
+      OmVolumeArgs omVolumeArgs) {
+    // Add default acls from volume.
+    List<OzoneAcl> acls = new ArrayList<>();
+    if (omBucketInfo.getAcls() != null) {
+      acls.addAll(omBucketInfo.getAcls());
+    }
+    omVolumeArgs.getAclMap().getDefaultAclList().forEach(
+        defaultAcl -> acls.add(
```

Comment:
Simple add does not merge acl properly. You can use OzoneAclUtil.inheritDefaultAcls(acls, omVolumeArgs.getAclMap().getDefaultAclList()) to do the merge.